lib/seattleflu/db/cli/command/etl/manifest.py

```diff
@@ -88,48 +90,170 @@ def etl_manifest(*, action: str):
+                    sample_barcode = manifest_record.document.pop("sample")
+                    sample_identifier = find_identifier(db, sample_barcode)
+
+                    if not sample_identifier:
+                        LOG.warning(f"Skipping sample with unknown sample barcode «{sample_barcode}»")
+                        mark_skipped(db, manifest_record.id)
+                        continue
+
+                    assert sample_identifier.set_name in expected_identifier_sets["samples"], \
+                        f"Sample identifier found in set «{sample_identifier.set_name}», not {expected_identifier_sets['samples']}"
+
+                    # Optionally, convert the collection barcode to full
+                    # identifier, ensuring it's known and from the correct
+                    # identifier set.
+                    collection_barcode = manifest_record.document.pop("collection", None)
+                    collection_identifier = find_identifier(db, collection_barcode) if collection_barcode else None
+
+                    if collection_barcode and not collection_identifier:
+                        LOG.warning(f"Skipping sample with unknown collection barcode «{collection_barcode}»")
+                        mark_skipped(db, manifest_record.id)
+                        continue
+
+                    assert not collection_identifier \
+                        or collection_identifier.set_name in expected_identifier_sets["collections"], \
+                            f"Collection identifier found in set «{collection_identifier.set_name}», not {expected_identifier_sets['collections']}"
+
+                    # Upsert sample cooperatively with enrollments ETL routine
+                    #
+                    # The details document was intentionally modified by two pop()s
+                    # earlier to remove barcodes that were looked up.  The
+                    # rationale is that we want just one clear place in the
+                    # warehouse for each piece of information.
+                    sample, status = upsert_sample(db,
+                        identifier            = sample_identifier.uuid,
+                        collection_identifier = collection_identifier.uuid if collection_identifier else None,
+                        additional_details    = manifest_record.document)
+                    
+                    mark_loaded(db, manifest_record.id,
                     status = status,
                     sample_id = sample.id)
+                
+                else:
+                    # Convert kit barcode to full identifier and check 
+                    # it's from the correct identifier set.
+                    kit_barcode = manifest_record.document.pop("sample")
+                    kit_identifier = check_barcode(db, 
+                        record          = manifest_record, 
+                        barcode         = kit_barcode, 
```

Comment:
There is trailing whitespace on some of the lines above and in other added lines as well (there's a "blank" line below which is all trailing whitespace). Trailing whitespace is generally considered undesirable (for various reasons, such as how it affects editor navigation), so it'd be nice to trim that. I believe your editor configuration options should include ways of automatically doing that (or at least showing you when it's present).

We should probably discuss using something automated around this and other code formatting styles.